src/mapping/device/GinkgoRBFKernels.cpp

```diff
@@ -1,132 +1,188 @@
+          }
+          const bool *deviceActiveAxis = activeAxis.data();
+
+          // Compute Euclidean distance using column-major indexing
+          for (size_t k = 0; k < activeAxis.size(); ++k) {
+            if (deviceActiveAxis[k]) {
+              double diff = k_supportPoints(k, j) - k_targetPoints(k, i);
+              dist += diff * diff;
+            }
+          }
+          dist        = Kokkos::sqrt(dist);
+          k_mtx(i, j) = f(dist, rbf_params); // Evaluate the RBF function
+        });
+  }
+}
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC4>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC4, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+template <typename EvalFunctionType>
+void create_rbf_system_matrix(std::shared_ptr<const gko::Executor>      exec,
+                              gko::ptr_param<GinkgoMatrix>              mtx,
+                              const std::array<bool, 3>                 activeAxis,
+                              gko::ptr_param<GinkgoMatrix>              supportPoints,
+                              gko::ptr_param<GinkgoMatrix>              targetPoints,
+                              EvalFunctionType                          f,
+                              ::precice::mapping::RadialBasisParameters rbf_params,
+                              bool                                      addPolynomial,
+                              unsigned int                              extraDims)
+{
+  if (std::dynamic_pointer_cast<const gko::ReferenceExecutor>(exec)) {
+    create_rbf_system_matrix_impl<Kokkos::HostSpace>(exec, mtx, activeAxis, supportPoints, targetPoints, f, rbf_params, addPolynomial, extraDims);
+  } else {
+    create_rbf_system_matrix_impl<Kokkos::DefaultExecutionSpace::memory_space>(exec, mtx, activeAxis, supportPoints, targetPoints, f, rbf_params, addPolynomial, extraDims);
+  }
+}
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC6>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC6, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+#define INSTATIATE_CREATE_RBF_SYSTEM_MATRIX(_function_type)                                                                                     \
```

Comment:
xSDK requires own macros to start with the project name. Also, typo :wink: 

```suggestion
#define PRECICE_INSTANTIATE_CREATE_RBF_SYSTEM_MATRIX(_function_type)                                                                                     \
```

Reply:
Yes, agree, I would probably even undefine it directly at the end of the file.